yt_dlp/extractor/mediastream.py

```diff
@@ -118,7 +122,18 @@ def _real_extract(self, url):
         formats, subtitles = [], {}
         for video_format in player_config['src']:
             if video_format == 'hls':
-                fmts, subs = self._extract_m3u8_formats_and_subtitles(player_config['src'][video_format], video_id)
+                params = {'at': 'web-app'}
+
+                for name, key in [('MDSTRMUID', 'uid'), ('MDSTRMSID', 'sid'), ('MDSTRMPID', 'pid'), ('VERSION', 'av')]:
+                    if val := self._search_regex(rf'window\.{name}\s*=\s*["\']([^"\']+)["\'];',
+                                                 webpage, key, default=None):
+                        params[key] = val
+
+                if access_token := parse_qs(url).get('access_token', [None])[0]:
+                    params['access_token'] = access_token
+
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                    update_url_query(player_config['src'][video_format], params), video_id)
```

Comment:
```suggestion
                params = {
                    'at': 'web-app',
                    'access_token': traverse_obj(parse_qs(url), ('access_token', 0)),
                }
                for name, key in (('MDSTRMUID', 'uid'), ('MDSTRMSID', 'sid'), ('MDSTRMPID', 'pid'), ('VERSION', 'av')):
                    params[key] = self._search_regex(
                        rf'window\.{name}\s*=\s*["\']([^"\']+)["\'];', webpage, key, default=None)

                fmts, subs = self._extract_m3u8_formats_and_subtitles(
                    update_url_query(player_config['src'][video_format], filter_dict(params)), video_id)
```

Seeing it in this reduced form made me realize all we wanted was filter_dict